tests/vlan/test_vlan.py

```diff
@@ -35,71 +33,162 @@ def cfg_facts(duthosts, rand_one_dut_hostname):
+    src_network = ipaddress.IPv4Interface(src_ipprefix).network
+    dst_network = ipaddress.IPv4Interface(dst_ipprefix).network
+    return src_network.overlaps(dst_network)
+
+
 @pytest.fixture(scope="module")
-def vlan_intfs_list():
-    return [ { 'vlan_id': vlan, 'ip': '192.168.{}.1/24'.format(vlan) } for vlan in vlan_id_list  ]
+def vlan_intfs_dict(cfg_facts, tbinfo):
+    vlan_intfs_dict = {}
+    for k, v in cfg_facts['VLAN'].items():
+        vlanid = v['vlanid']
+        for addr in cfg_facts['VLAN_INTERFACE']['Vlan'+vlanid]:
+            if addr.find(':') == -1:
+                ip = addr
+                break
+        else:
+            continue
+        logger.info("Original VLAN {}, ip {}".format(vlanid, ip))
+        vlan_intfs_dict[int(vlanid)] = {'ip': ip, 'orig': True}
+    # For t0 topo, will add 2 VLANs for test.
+    # Need to make sure vlan id is unique, and avoid vlan ip network overlapping.
+    # For example, ip prefix is 192.168.0.1/21 for VLAN 1000,
+    # Below ip prefix overlaps with 192.168.0.1/21, and need to skip:
+    # 192.168.0.1/24, 192.168.1.1/24, 192.168.2.1/24, 192.168.3.1/24,
+    # 192.168.4.1/24, 192.168.5.1/24, 192.168.6.1/24, 192.168.7.1/24
+    if tbinfo['topo']['name'] != 't0-56-po2vlan':
+        vlan_cnt = 0
+        for i in xrange(0, 255):
+            vid = 100 + i
+            if vid in vlan_intfs_dict:
+                continue
+            ip = u'192.168.{}.1/24'.format(i)
+            for v in vlan_intfs_dict.values():
+                if compare_network(ip, v['ip']):
+                    break
+            else:
+                logger.info("Add VLAN {}, ip {}".format(vid, ip))
+                vlan_intfs_dict[vid] = {'ip': ip, 'orig': False}
+                vlan_cnt += 1
+            if vlan_cnt >= 2:
+                break
+    return vlan_intfs_dict
 
 
 @pytest.fixture(scope="module")
-def vlan_ports_list(rand_selected_dut, tbinfo, cfg_facts):
+def vlan_ports_list(duthosts, rand_one_dut_hostname, rand_selected_dut, tbinfo, cfg_facts, vlan_intfs_dict):
+    duthost = duthosts[rand_one_dut_hostname]
     mg_facts = rand_selected_dut.get_extended_minigraph_facts(tbinfo)
     vlan_ports_list = []
     config_ports = {k: v for k,v in cfg_facts['PORT'].items() if v.get('admin_status', 'down') == 'up'}
     config_portchannels = cfg_facts.get('PORTCHANNEL', {})
     config_port_indices = {k: v for k, v in mg_facts['minigraph_ptf_indices'].items() if k in config_ports}
     ptf_ports_available_in_topo = {port_index: 'eth{}'.format(port_index) for port_index in config_port_indices.values()}
-    config_port_channel_members = [port_channel[1]['members'] for port_channel in config_portchannels.items()]
+    config_port_channel_members = [port_channel['members'] for port_channel in config_portchannels.values()]
     config_port_channel_member_ports = list(itertools.chain.from_iterable(config_port_channel_members))
+    # key is dev name, value is list for configured VLAN member.
```

Comment:
># key is dev name, value is list for configured VLAN member. [](http://example.com/codeflow?start=4&length=60)

Do you mean `k` `v` in below for-loop? If yes, could you move comment line just above for-loop?